Userland/Libraries/LibWeb/Loader/Resource.h

```diff
@@ -52,7 +52,8 @@ class Resource : public RefCounted<Resource> {
     void register_client(Badge<ResourceClient>, ResourceClient&);
     void unregister_client(Badge<ResourceClient>, ResourceClient&);
 
-    const String& encoding() const { return m_encoding; }
+    bool has_encoding() const { return m_encoding.has_value(); }
+    const Optional<String> encoding() const { return m_encoding; }
```

Comment:
This should be returned by reference to avoid copies.
```suggestion
    const Optional<String>& encoding() const { return m_encoding; }
```